packages/io/dxf-deserializer/instantiate.js

```diff
@@ -207,79 +221,29 @@ function instantiateEllipse (obj, layers, options) {
   let sptz = obj['sptz']
   let swid = obj['swid'] // Ratio of minor axis to major axis
   // FIXME need to determine resolution from object/layer/variables
-  let res = CSG.defaultResolution2D
+  let res = 16
 
   // convert to 2D object
   if (pptz === 0.0 && sptz === 0.0) {
-    let center = new CSG.Vector2D(0, 0)
-    let mjaxis = new CSG.Vector2D(sptx, spty)
-    let rx = center.distanceTo(mjaxis)
+    let center = math.vec2.fromValues(0, 0)
+    let mjaxis = math.vec2.fromValues(sptx, spty)
+    let rx = math.vec2.distance(center, mjaxis)
     let ry = rx * swid
     let angle = Math.atan2(spty, sptx) * 180 / Math.PI
-    if (angle < CSG.EPS) angle = 0
+    if (angle < EPS) angle = 0
+    angle = angle * 0.017453292519943295 // radians
+
     // FIXME add start and end angle when supported
-    let cag = CAG.ellipse({center: [0,0], radius: [rx, ry], resolution: res}).rotateZ(angle).translate([pptx,ppty])
-    return cag
+    let cag = primitives.ellipse({ center: [0, 0], radius: [rx, ry], segments: res })
+    let matrix = math.mat4.fromZRotation(angle)
+    matrix = math.mat4.multiply(matrix, math.mat4.fromTranslation([pptx, ppty, 0]))
+    // cag.rotateZ(angle).translate([pptx,ppty])
+    return geometry.geom2.transform(matrix, cag)
   }
   // convert to 3D object
 }
 
-function createEdges (vlen, faces) {
-  let edges = []
-  while (vlen > 0) {
-    edges.push([])
-    vlen--
-  }
-  let mod3 = Math.floor(faces.length / 3) * 3
-  if (mod3 === faces.length) {
-    let fi = 0
-    while (fi < faces.length) {
-      let v1 = faces[fi++]
-      let v2 = faces[fi++]
-      let v3 = faces[fi++]
-      if (v1 === v2 || v1 === v3 || v2 === v3) continue
-
-      let edge = edges[v1]
-      if (edge.indexOf(v2) < 0) { edge.push(v2) }
-      if (edge.indexOf(v3) < 0) { edge.push(v3) }
-
-      edge = edges[v2]
-      if (edge.indexOf(v3) < 0) { edge.push(v3) }
-      if (edge.indexOf(v1) < 0) { edge.push(v1) }
-
-      edge = edges[v3]
-      if (edge.indexOf(v1) < 0) { edge.push(v1) }
-      if (edge.indexOf(v2) < 0) { edge.push(v2) }
-    }
-  }
-  return edges
-}
-
-function createFaces (edgesByVertex) {
-  let v1 = edgesByVertex.length
-  let faces = []
-  while (v1 > 0) {
-    v1--
-    let v1edges = edgesByVertex[v1]
-    let e1i = v1edges.length
-    while (e1i > 0) {
-      e1i--
-      let v2 = v1edges[e1i]
-      let v2edges = edgesByVertex[v2]
-      // search for common vertexes
-      let e2i = v2edges.length
-      while (e2i > 0) {
-        e2i--
-        let v3 = v2edges[e2i]
-        if (v1edges.indexOf(v3) < 0) continue
-        faces.push([v1, v2, v3])
-      }
-    }
-  }
-  return faces
-}
-
-function instantiateFaces (fvals) {
+const instantiateFaces = (fvals) => {
```

Comment:
no need for the rest anymore ?

Reply:
Both createEdges() and createFaces() are unused. So.. bye bye!

Reply:
:) yeaah !!